walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,57 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Base cases
+    if (request.getAttendees().isEmpty()) { return Arrays.asList(TimeRange.WHOLE_DAY); }
+    if (request.getDuration() >= TimeRange.WHOLE_DAY.duration()) { return new ArrayList<>(); }
+    
+    // Preprocess event input
+    List<Event> pertinentEvents = events.stream()
+                                        .filter(event -> eventContainsAttendees(event, new HashSet<>(request.getAttendees())) && 
+                                                         eventCleared(new ArrayList<>(events), event))
+                                        .sorted((e1, e2) -> TimeRange.ORDER_BY_END.compare(e1.getWhen(), e2.getWhen()))
+                                        .collect(Collectors.toList());
```

Comment:
The format seems weird here. Did you use auto formatter?

Reply:
I couldn't really find a style guide section that specified how this was supposed to be formatted (line-wrapping and horizontal alignment sections both were iffy on mandatory styling) so I did it in a way that I thought was logical: line up the statements that are of the same depth. Is there another standard way to align?